ebundala/free-mentors
# #168143705 Create a signup Rest endpoint

#### What does this PR do?
Create a signup endpoint
```sh
Post /auth/signup
```
#### Description of Task to be completed?
create a REST endpoint to allow user signup
#### How should this be manually tested?
to test manualy this endpoint using curl copy this commandsl and run on your terminal
```sh
curl -X POST "https://freementorsx.herokuapp.com/api/v1/auth/signup" -H "accept: application/json" -H "Content-Type: application/json" -d "{\"id\":0,\"firstName\":\"string\",\"lastName\":\"string\",\"email\":\"string@email.com\",\"password\":\"string\",\"address\":\"string\",\"bio\":\"string\",\"occupation\":\"string\",\"expertise\":\"string\",\"gender\":\"string\",\"dob\":\"string\",\"avator\":\"string\",\"role\":\"mentee\"}"

```
expected result
```json
{
  "status": 201,
  "data": {
    "firstName": "string",
    "lastName": "string",
    "address": "string",
    "bio": "string",
    "occupation": "string",
    "expertise": "string",
    "email": "string@email.com",
    "dob": "string",
    "gender": "string",
    "password": null,
    "role": "mentee",
    "createdOn": "Thu, 29 Aug 2019 23:08:38 GMT",
    "id": 1567120118113
  },
  "message": "User created successfully"
}
```
#### What are the relevant pivotal tracker stories?
[#168143705](https://www.pivotaltracker.com/story/show/168143705)
